Reports: add category name to criteria list in Manage Criteria

Harrison-Idornigie opened the pull request:
<!-- Provide a brief summary of your changes in the Pull Request title.
Be sure to check out our contributor docs for more info about pull requests.
https://github.com/GibbonEdu/core/blob/master/docs/CONTRIBUTING.md#how-to-submit-a-pull-request -->

**Description**
<!-- Please describe your changes in detail. -->

**Motivation and Context**
<!-- Why is this change suggested? Is there a problem it helps to solve?
If this PR fixes an open issue, please link to the issue here. -->

**How Has This Been Tested?**
<!-- Please describe how you've tested your changes. Include details of 
your testing environment, and any tests you've run to see how your 
change affects other areas of the code. -->

**Screenshots**
<!-- If applicable, add screenshots to help illustrate your changes. -->

SKuipers commented on modules/Reports/reporting_criteria_manage.php:
```diff
@@ -188,6 +188,10 @@
 
     $table->addColumn('name', __('Criteria'));
     $table->addColumn('criteriaType', __('Type'));
+    $table->addColumn('category', __('Category'))
+        ->format(function ($row) {
+            return $row['category'] ?? '';
+        });
```
Rather than adding a new column which could take up more horizontal space, especially if category is not used by a school, lets use a formatDetails to combine this with the Type column. What do you think?

```suggestion
    $table->addColumn('criteriaType', __('Type'))
        ->description(__('Category'))
        ->formatDetails(function ($row) {
            return $row['category'] ?? '';
        });
```

Harrison-Idornigie commented:
> Thanks for the PR, looks like a good idea based on your previous changes. I've added a comment to the code. Also, we've just fixed the breaking tests in v29, so can you rebase your branch onto the current commit of v29. Thanks! 😃

Thanks for your response. We just wanted to be able to visually categorize the report fields and it makes the generated reports look well formatted and organized

Harrison-Idornigie commented:
@SKuipers I need advice on how to add student number generation logic, similar to userID generation. We specify the format and when the students are being registered, the next number in the sequence is added for the student if no unique value is provided. 

Or at least separate the userID generation logic for students and staff. We could add the student ID generation logic here:
![image](https://github.com/user-attachments/assets/1092862f-2f75-4db3-9f52-d320d6b2c339)

Or I was thinking of using a custom module but I'm not sure how that would work.

SKuipers commented:
Hi @Harrison-Idornigie I think what you're looking for may already exist. If you go into User Admin > User Settings, at the top you can define Username Formats separately for different roles, including prefixes and numerical counting. These usernames could also be used as student IDs, so that both are unique and auto-generated by the system.
